src/Raven.Server/Documents/Replication/DocumentReplicationExecuter.cs

```diff
@@ -16,69 +18,110 @@
 namespace Raven.Server.Documents.Replication
 {
     //TODO : add DocumentReplicationStatistics that will track operational data
-    //TODO : do not forget to handle authentication stuff
     public class DocumentReplicationExecuter: BaseReplicationExecuter
     {
-        private readonly ReplicationWebSocket _socket;
+        private readonly DocumentReplicationTransport _transport;
         private readonly Guid _srcDbId;
         private readonly ReplicationDestination _destination;
-        private long _lastSentEtag;
+        private long _lastEtag;
+
+        
 
         public DocumentReplicationExecuter(DocumentDatabase database, 
             Guid srcDbId, 
-            ReplicationDestination destination, 
-            long lastSentEtag) : base(database)
+            ReplicationDestination destination) : base(database)
         {
             _srcDbId = srcDbId;
             _destination = destination;
-            _lastSentEtag = lastSentEtag;
-            _socket = new ReplicationWebSocket(destination.Url);		
+            _transport = new DocumentReplicationTransport(destination.Url,_database.DatabaseShutdown);
+            DocumentsOperationContext context;
+            _database.DocumentsStorage.ContextPool.AllocateOperationContext(out context);
+
         }
 
         //if _destination == null --> the replication is only incoming, otherwise it is a two-way one
         public override string Name => _srcDbId + (_destination != null ? $"/{_destination.Url}" : string.Empty);
 
+        public Guid SrcDatabasebId => _srcDbId;
+
         //by design this method won't handle opening and commit of the transaction
         //(that should happen at the calling code)
         public void ReceiveReplicatedDocuments(DocumentsOperationContext context,
             List<BlittableJsonReaderObject> docs)
         {
             var dbChangeVector = _database.DocumentsStorage.GetChangeVector(context);
             var changeVectorUpdated = false;
+            var maxReceivedChangeVector = new Dictionary<Guid,long>();
             for(int i = 0; i < docs.Count; i++)
             {
                 var doc = docs[i];
-                changeVectorUpdated = UpdateDbChangeVectorIfNeededFrom(doc,dbChangeVector);
+                var changeVector = doc.EnumerateChangeVector();
+                for (int j = 0; j < changeVector.Length; j++)
+                {
+                    var currentEntry = changeVector[j];
+                    long existingValue;
+                    if(!maxReceivedChangeVector.TryGetValue(currentEntry.DbId,out existingValue) ||
```

Comment:
The usage of `maxReceivedChangeVector` here is confusing. You are using indexer and add to effectively do the same thing. Why?